pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -341,40 +349,34 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 			cidrStrings = append(cidrStrings, ipv6Addr.String())
 		}
 	} else {
-		// multi-networking enabled clusters
-		cidrStrings, err = ca.performMultiNetworkCIDRAllocation(node, instance.NetworkInterfaces)
-		if err != nil {
-			nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
-			return fmt.Errorf("failed to get cidr(s) from provider: %v", err)
+		if hasNodeLabels {
+			cidrStrings = ca.getDefaultNwCIDRs(node, instance.NetworkInterfaces, defaultSubnet, defaultPodRange)
 		}
 	}
-	if len(cidrStrings) == 0 {
-		nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
-		return fmt.Errorf("failed to allocate cidr: Node %v has no CIDRs", node.Name)
-	}
-	// Can have at most 2 ips (one for v4 and one for v6)
-	if len(cidrStrings) > 2 {
-		klog.InfoS("Got more than 2 ips, truncating to 2", "cidrStrings", cidrStrings)
-		cidrStrings = cidrStrings[:2]
-	}
 
-	cidrs, err := netutils.ParseCIDRs(cidrStrings)
-	if err != nil {
-		return fmt.Errorf("failed to parse strings %v as CIDRs: %v", cidrStrings, err)
+	// update Node.Spec.PodCIDR(s) right away with cidrStrings
+	// for clusters 1) without multi-networking, and 2) with multi-networking and node labels("new way")
+	if multiNetworkingDisabled || !multiNetworkingDisabled && hasNodeLabels {
+		if err = ca.updateNodePodCIDRWithCidrStrings(oldNode, node, cidrStrings); err != nil {
+			return err
+		}
 	}
-	if len(cidrs) > 1 {
-		if dualStack, _ := netutils.IsDualStackCIDRs(cidrs); !dualStack {
-			return fmt.Errorf("err: IPs are not dual stack, CIDRS: %v", cidrStrings)
+	// performing multiNetworkCIDRAllocation for multi-networking clusters
+	if !multiNetworkingDisabled {
+		cidrStrings, err = ca.performMultiNetworkCIDRAllocation(node, instance.NetworkInterfaces, hasNodeLabels)
+		if err != nil {
+			nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
+			return fmt.Errorf("failed to get cidr(s) from provider: %v", err)
+		}
+		// update Node.Spec.PodCIDR(s) after performing cidr allocation for
+		// 3) clusters with multi-networking and without node labels("old way")
+		if !hasNodeLabels {
+			if err = ca.updateNodePodCIDRWithCidrStrings(oldNode, node, cidrStrings); err != nil {
+				return err
+			}
 		}
 	}
 
-	node.Spec.PodCIDR = cidrStrings[0]
-	node.Spec.PodCIDRs = cidrStrings
-
-	err = ca.updateNodeCIDR(node, oldNode)
-	if err != nil {
-		return err
-	}
```

Comment:
Any reason why you are not using this function anymore? You could have structured the existing code to continue using this call right?

Reply:
Wrapped in a function below.